apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProviderIT.java

```diff
@@ -0,0 +1,1758 @@
+      snfRifRecord.setDiagnosis2Code(
+          Optional.of(SamhsaMatcherTest.SAMPLE_SAMHSA_ICD_9_DIAGNOSIS_CODE));
+      snfRifRecord.setDiagnosis2CodeVersion(Optional.of('9'));
+      entityManager.merge(snfRifRecord);
+      entityManager.getTransaction().commit();
+    } finally {
+      if (entityManager.getTransaction().isActive()) entityManager.getTransaction().rollback();
+      if (entityManager != null) entityManager.close();
+      if (entityManagerFactory != null) entityManagerFactory.close();
+    }
+
+    IGenericClient fhirClient = ServerTestUtils.createFhirClientV2();
+
+    Bundle searchResults =
+        fhirClient
+            .search()
+            .forResource(ExplanationOfBenefit.class)
+            .where(
+                ExplanationOfBenefit.PATIENT.hasId(
+                    TransformerUtilsV2.buildPatientId(carrierRifRecord.getBeneficiaryId())))
+            .and(new StringClientParam("excludeSAMHSA").matches().value("true"))
+            .returnBundle(Bundle.class)
+            .execute();
+    Assert.assertNotNull(searchResults);
+    for (ClaimTypeV2 claimType : ClaimTypeV2.values()) {
+      /*
+       * First, verify that the claims that should have been filtered out, were. Then
+       * in the `else` clause, verify that everything was **not** filtered out.
+       */
+      // FIXME remove the `else if`s once filtering fully supports all claim types
+      if (claimType.equals(ClaimTypeV2.CARRIER))
+        Assert.assertEquals(0, filterToClaimType(searchResults, claimType).size());
+      else if (claimType.equals(ClaimTypeV2.HHA))
+        Assert.assertEquals(0, filterToClaimType(searchResults, claimType).size());
+      else if (claimType.equals(ClaimTypeV2.HOSPICE))
+        Assert.assertEquals(0, filterToClaimType(searchResults, claimType).size());
+      else if (claimType.equals(ClaimTypeV2.INPATIENT))
+        Assert.assertEquals(0, filterToClaimType(searchResults, claimType).size());
+      else if (claimType.equals(ClaimTypeV2.OUTPATIENT))
+        Assert.assertEquals(0, filterToClaimType(searchResults, claimType).size());
+      else if (claimType.equals(ClaimTypeV2.SNF))
+        Assert.assertEquals(0, filterToClaimType(searchResults, claimType).size());
+      else if (claimType.equals(ClaimTypeV2.PDE))
+        // PDE Claims do not contain SAMHSA fields and thus won't be filtered.
+        Assert.assertEquals(1, filterToClaimType(searchResults, claimType).size());
+      else Assert.assertEquals(1, filterToClaimType(searchResults, claimType).size());
+    }
+  }
+
+  /**
+   * Verifies that {@link
+   * gov.cms.bfd.server.war.r4.providers.ExplanationOfBenefitResourceProvider#findByPatient(ca.uhn.fhir.rest.param.ReferenceParam)}
+   * handles the {@link ExplanationOfBenefitResourceProvider#HEADER_NAME_INCLUDE_TAX_NUMBERS} header
+   * properly.
+   *
+   * @throws FHIRException (indicates test failure)
+   */
+  // TODO: Fix this test .. it isn't working. Tax number is not showing up.
```

Comment:
Someone else will need to look at this.  I couldn't figure out why TAX Number wasn't being handled here.

Reply:
done